### ISSUE TITLE
Formatting in annotation formats is not preserved

### ISSUE THREAD

**Issue by Bidski** [bit-bots/imagetagger#46]
When exporting annotations the indentation of each line is not preserved. This breaks formats that rely on indentation to work (like YAML).

For example, a format like this:
```
# Base format
imageset_%%imageset:
    description: %%setdecsription
    team: %%team
    location: %%setlocation
    %%content

# Image format
    %%imagename:
        dimensions: [%%imagewidth, %%imageheight]
        num_annotations: %%annoamount
        %%annotations
```

Produces an output like this:
```
imageset_bitbots-nagoya-01:
    description: %%setdecsription
    team: Bit-Bots
    location: nagoya
    
ball_move_1_00001.png:
        dimensions: [800, 600]
        num_annotations: 5
```

Here, the issue is that `ball_move_1_00001.png:` should be indented to line up with `location`. The same issue occurs with the annotation formats too.

It also appears as though an extra newline is appended to the base format. While this shouldn't break  YAML syntax, it is an unexpected occurrence.

**Comment by NFiedler** [comment]
We actually had the same problem before, but we fixed it by adding the indentation to the image and annotation formats manually in the admin console. Now i tried to create the right indentation in the regular form offered to the user, but the indentation somehow gets lost in the process. While this shouldn't happen i'd additionally propose ```%%indentedcontent``` as a content placeholder which preserves the indentation of the placeholder.

The added newline is a weird quirk, I'll look into it.

**Comment by NFiedler** [comment]
https://stackoverflow.com/questions/38995764/django-textfield-charfield-is-stripping-spaces-blank-lines

This looks to be part of the problem.

**Comment by NFiedler** [comment]
Now the white-spaces at the beginning of the text area are kept. To indent the annotations for example, the indentation has to be made in the annotation fornat for every single line. **But** the %%annotation placeholder shouldn't be indented in the image format.
This is not the perfect solution for this issue, just a way around...

**Comment by johagge** [comment]
Currently white-spaces are still stripped in the Vector format.

**Comment by timonegk** [comment]
@johagge I fixed that in f3d9bcb31a83366a85758d99db03ef29853260ad.

**Comment by jaagut** [comment]
So, this issue can be closed?
